internal/provider/resource_gitlab_project.go

```diff
@@ -745,6 +744,10 @@ func resourceGitlabProjectSetToState(client *gitlab.Client, d *schema.ResourceDa
 	d.Set("wiki_access_level", string(project.WikiAccessLevel))
 	d.Set("squash_commit_template", project.SquashCommitTemplate)
 	d.Set("merge_commit_template", project.MergeCommitTemplate)
+
+	//Note: This field is deprecated and will always be nil starting in GitLab 15.0.
+	d.Set("build_coverage_regex", project.BuildCoverageRegex)
```

Comment:
This is my java background showing through again :( , you're absolutely right it will be an empty string. I'll update the comment, but it's still probably a good comment.

Reply:
I appreciate the comment 🎉